tests/keyboard/scancode/test_unicode_to_scancode.cpp

```diff
@@ -5,297 +5,141 @@
 namespace
 {
     using namespace scancode;
-
-    Scancode16bits_t get_and_pop_elem(ScancodeSeq_t& scancode_seq)
-    {
-        Scancode16bits_t scancode = scancode_seq.front();
-
-        scancode_seq.pop();
-        return scancode;
-    }
-
-
+    
+    
     class FrenchLocaleFixture
     {
       protected :
-        static constexpr int _LCID = 0x0000040c;
+        static constexpr int LCID = 0x0000040c;
     };
+
+    
+#define UNI_TO_SCAN_TEST_INIT(unicode)                                  \
+    ScancodeSeq scancode_seq = unicode_to_scancode(LCID, unicode);      \
+    array_view<Scancode16bits> scancodes = scancode_seq.scancodes();    \
+
+#define UNI_TO_SCAN_TEST_EXPECT(expected_size, ...)                 \
+    RED_REQUIRE_EQUAL(scancodes.size(), expected_size);             \
+                                                                    \
+    constexpr std::uint16_t EXPECTED_SEQ[] = {__VA_ARGS__};         \
+                                                                    \
+    RED_CHECK_EQUAL(scancodes, make_array_view(EXPECTED_SEQ));      \
```

Comment:
The size check is already done by comparing the array_view. Better, it also displays values: one size may be different AND the data may not match at all, only size is not a sufficient diagnosis.